airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java

```diff
@@ -120,11 +121,13 @@ static class BufferManagerDequeue {
 
     private final GlobalMemoryManager memoryManager;
     private final Map<StreamDescriptor, MemoryBoundedLinkedBlockingQueue<AirbyteMessage>> buffers;
+    private final Map<StreamDescriptor, ReentrantLock> bufferLocks;
 
     public BufferManagerDequeue(final GlobalMemoryManager memoryManager,
                                 final Map<StreamDescriptor, MemoryBoundedLinkedBlockingQueue<AirbyteMessage>> buffers) {
       this.memoryManager = memoryManager;
       this.buffers = buffers;
+      bufferLocks = new HashMap<>();
```

Comment:
Does this need to be a `ConcurrentHashMap`?

Reply:
i think actually it probably should. 2 threads could try to put the lock at once.

Reply:
buffers should also be a concurrenthasmap